src/us/kbase/auth2/service/AuthBuilder.java

```diff
@@ -67,10 +69,20 @@ private MongoClient buildMongo(final AuthStartupConfig c) throws StorageInitExce
 				final MongoCredential creds = MongoCredential.createCredential(
 						c.getMongoUser().get(), c.getMongoDatabase(), c.getMongoPwd().get());
 				// unclear if and when it's safe to clear the password
-				return new MongoClient(new ServerAddress(c.getMongoHost()), creds,
-						MongoClientOptions.builder().build());
+				return MongoClients.create(
+						MongoClientSettings.builder()
+								.credential(creds)
+								.applyToClusterSettings(builder ->
+										builder.hosts(Arrays.asList(
+												new ServerAddress(c.getMongoHost()))))
+								.build());
 			} else {
-				return new MongoClient(new ServerAddress(c.getMongoHost()));
+				return MongoClients.create(
+						MongoClientSettings.builder()
+								.applyToClusterSettings(builder ->
+										builder.hosts(Arrays.asList(
+												new ServerAddress(c.getMongoHost()))))
+								.build());
```

Comment:
I would create the `MongoClientSettings` builder prior to the `if`, and in the no creds case just `.build()` and in the creds case `.credential(creds).build()`

Reply:
👍

Reply:
:+1: